examples/checkbox_toppings.py

```diff
@@ -4,7 +4,10 @@
 if __name__ == "__main__":
     toppings = (
         questionary.checkbox(
-            "Select toppings", choices=["foo", "bar", "bazz"], style=custom_style_dope
+            "Select toppings",
+            choices=["foo", "bar", "bazz"],
+            validate=lambda a: (len(a) != 0, "You must select at least one topping"),
```

Comment:
I think the API isn't ideal as there are three different types of arguments for the validate argument accross the different prompts:
- function returning one boolean
- function returning a tuple (predicated + error message)
- prompt toolkit validator 

Since it would be a bit more effort to unify this properly I am good with this for the moment.

Reply:
@tmbo I'm happy to sort this before I merge. Do you think implementing it using the `Validator` from `prompt_toolkit` would be better?

Reply:
I don't think we can use the `Validator` from `prompt_toolkit`: its `validate` function takes a `document` of type `prompt_toolkit.document.Document` as its argument, which we don't have in this case. This seems we'd need to roll our own class (and in terms of backwards compatibility, support the prompt toolkit validator as well).

Reply:
@tmbo A document instance can be created as so, `prompt_toolkit.document.Document("Some text")`, which might be sufficient?

This PR would actually probably have been better submitted to https://github.com/prompt-toolkit/python-prompt-toolkit to fix https://github.com/prompt-toolkit/python-prompt-toolkit/issues/430, which we could then have used here. What would you prefer?

Reply:
I mean if upstream prompt toolkit is open to improve this on their end this would be even better 🙂 otherwise I think trying the `Document` creation might actually work, I didn't have time to take a closer look if there is something else happing when creating a `Document`, but if there are no side effects that prevent other functionality from working I think we are good

Reply:
@tmbo I will implement the `Document` method for now. If `prompt_toolkit` accept the change, then as we are using the `Document` object, the API for questionary will probably not change, so it would simply be a case of changing how it is implemented.

Reply:
@tmbo After looking at this more closely, I can't use a `Document` because it takes a string, and we need to pass an array to the validator.

If you are happy with it in its current form then I will merge - we can look at tidying up the interface later.

Reply:
yes fine with me :+1: